symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -1804,8 +1803,71 @@ void test_matrix()
     basic_free_stack(i4);
 }
 
+void test_lambda_double()
+{
+    basic two, x, y, z, r, s;
+    basic_new_stack(two);
+    basic_new_stack(x);
+    basic_new_stack(y);
+    basic_new_stack(z);
+    basic_new_stack(r);
+    basic_new_stack(s);
+    CVecBasic *args = vecbasic_new();
+    CVecBasic *exprs = vecbasic_new();
+
+    integer_set_si(two, 2);
+    symbol_set(x, "x");
+    symbol_set(y, "y");
+    symbol_set(z, "z");
+    symbol_set(r, "r");
+    symbol_set(s, "s");
+
+    vecbasic_push_back(args, x);
+    vecbasic_push_back(args, y);
+    vecbasic_push_back(args, z);
+
+    // r = x + y*z + (y*z)**2
+    // s = 2*x + y*z + (y*z)**2
+    basic_mul(y, y, z);
+    basic_pow(z, y, two);
+    basic_add(z, z, y);
+    basic_add(r, x, z);
+    basic_mul(x, two, x);
+    basic_add(s, x, z);
+    vecbasic_push_back(exprs, r);
+    vecbasic_push_back(exprs, s);
+
+    double outs[2];
+    double inps[3] = {1.5, 2.0, 3.0};
+    for (int perform_cse = 0; perform_cse <= 1; ++perform_cse) {
```

Comment:
This fails with ` error: ‘for’ loop initial declarations are only allowed in C99 mode`

Reply:
Got it, fixing. (though I thought C89 required *all* variables to be declared at top of function which is not the case in many other tests)